patch/1.21.4.1/lua-resty-core-tlshandshake.patch

```diff
@@ -178,44 +178,65 @@ index 0000000..89454ad
 +
 +    local client_cert, client_pkey
 +
-+    local client_cert_path = options.client_cert_path
-+    local client_pkey_path = options.client_priv_key_path
-+    if client_cert_path then
-+        if not client_pkey_path then
++    if options.client_cert_path or options.client_cert then
++        if options.client_cert_path and options.client_cert then
++            error("client client_cert_path and client_cert both setting ", 2)
++        end
++
++        if not options.client_priv_key_path and not options.client_priv_key then
 +            error("client certificate supplied without corresponding " ..
-+                  "private key", 2)
++                    "private key", 2)
 +        end
 +
-+        if type(client_cert_path) ~= "string" then
-+            error("bad client_cert option type", 2)
++        if options.client_priv_key_path and options.client_priv_key then
++            error("client certificate private key supplied with " ..
```

Comment:
The error message is somewhat ambiguous.
It should be noted that: **Only one of the parameters 'a' and 'b' can be set**
WDYT?

Reply:
APISIX's ci checks the error message format returned by this function.